src/Servers/Kestrel/Transport.Quic/src/Internal/QuicStreamContext.cs

```diff
@@ -402,17 +397,6 @@ private async ValueTask ShutdownWrite(Exception? shutdownReason)
 
                     _stream.Shutdown();
                 }
-
-                try
-                {
-                    // TODO: Take cancellation token?
-                    await _stream.ShutdownCompleted();
```

Comment:
ShutdownCompleted waits until the client stream also shuts down. I'm pretty sure we want to clean up the request once we've finished sending it, and not hold onto everything until the client is done.

I'm double-checking with QUIC team that this is ok.